File: leetcode/2391-minimum-amount-of-time-to-collect-garbage/garbageCollection.go
package leetcode

func garbageCollection(garbage []string, travel []int) int {
	var minutes = 0
	var (
		garbageMinutes    = 0
		travelMinutesMap  = make(map[rune]int)
		temptravelMinutes = 0
	)

	for index, garbageItems := range garbage {

		garbageMinutes = garbageMinutes + len(garbageItems)

		if index > 0 {
			temptravelMinutes = temptravelMinutes + travel[index-1]
		}

		for _, garbageItem := range garbageItems {
			travelMinutesMap[garbageItem] = temptravelMinutes
		}
	}

	minutes = minutes + garbageMinutes

	for _, travelMinutes := range travelMinutesMap {
		minutes = minutes + travelMinutes
	}

	return minutes
}

File: leetcode/2391-minimum-amount-of-time-to-collect-garbage/garbageCollection.ts
function garbageCollection(garbage: string[], travel: number[]): number {
  let minutes = 0;
  // 单独收拾垃圾的时间， 就是垃圾的数量
  let garbageMinutes = 0;
  // 每个垃圾车开车需要的时间
  const travelMinutesMap = new Map<string, number>();
  // 临时变量
  let temptravelMinutes = 0;

  for (const [index, garbageItems] of garbage.entries()) {
    garbageMinutes = garbageMinutes + garbageItems.length;

    if (index > 0) {
      temptravelMinutes = temptravelMinutes + travel[index - 1];
    }

    for (const garbageItem of garbageItems) {
      travelMinutesMap.set(garbageItem, temptravelMinutes);
    }
  }

  minutes = minutes + garbageMinutes;

  for (const [_, travelMinutes] of travelMinutesMap) {
    minutes = minutes + travelMinutes;
  }

  return minutes;
}

export default garbageCollection;
